ophyd/pv_positioner.py

```diff
@@ -296,3 +301,151 @@ def done_moving(**kwargs):
         else:
             self.setpoint.put(position, wait=False,
                               callback=done_moving)
+
+
+class PVPositionerComparator(PVPositioner):
+    """
+    PV Positioner with a software done signal.
+
+    The done state is set by a comparison function defined in the class body.
+    The comparison function takes two arguments, readback and setpoint,
+    returning True if we are considered done or False if we are not.
+
+    This class is intended to support `PVPositionerIsClose`, but exists to
+    allow some flexibility if we want to use other metrics for deciding if
+    the PVPositioner is done.
+
+    Internally, this will subscribe to both the `setpoint` and `readback`
+    signals, updating `done` as appropriate.
+    """
```

Comment:
Should add the parameters to the docstring here.

Reply:
I when-in-rome'd a bit too much here- a lot of the surrounding code was also missing parameters. I'll update.